### DISCUSSION TITLE
`repository_advisory` event not getting delivered to GitHub App

### DISCUSSION THREAD

**Discussion by sethmlarson** [community/community#67518] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

Could be a fault on my part, but I've tried many different configurations and permissions and simply can't get an event of type `repository_advisory` to be delivered to my GitHub App with this minimal configuration:

## Reproducing steps

- Create a private GitHub App with the following config:
  - Permissions:
    - Repository security advisories: `read and write`
  - Subscribe to Events:
    - Repository advisory `[x]`
- Install the GitHub App
  - Enable for all repositories
- Create a draft security advisory on a repository
- See that no event is fired in the GitHub app event logs (expecting `repository_advisory.reported`)
- Reconfigure the app to subscribe to star events
- Star the same repository.
- See the app event logs reporting, and that the app server receives the `star.created` event.

Is there something that I'm missing or is this genuinely a bug in GitHub App event delivery?

**Comment by captn3m0** [comment, marked as answer]
The [event on the docs](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=reported#repository_advisory) site says:

> A private vulnerability report was submitted.

While you're trying to trigger it by "Create a draft security advisory on a repository". 

You need to enable [Private Vulnerability Reporting](https://docs.github.com/en/code-security/security-advisories/working-with-repository-security-advisories/configuring-private-vulnerability-reporting-for-a-repository#enabling-or-disabling-private-vulnerability-reporting-for-a-repository) on the repo, and then create a vulnerability report with a second account to actually trigger the "reported" event.

The report submission form is not available to accounts with write access. (Could be admin access, its unclear). You can see a sample form at https://github.com/captn3m0/playtube/security/advisories/new

If you instead publish the advisory, this would trigger the `repository_advisory.published` event, which is probably what you want. There is no event for a "draft" advisory creation. Drafts in general are considered private throughout GitHub (such as with draft releases), and there aren't any events fired for those.

**Reply by sethmlarson** [reply]
@captn3m0 Thanks for the reply, I do believe I've done everything you mention and have created many draft advisories (admittedly using my @sethmlarson account, does it matter which account is used to determine whether events are fired? I wouldn't expect it to...)

[Here's the repository](https://github.com/sethmlarson/ghsa-default-collabs), if you had a second could you file a report?

**Reply by sethmlarson** [reply]
@captn3m0 Wow, it totally does matter who does the reporting! The report you created sent the report into the "Triage" state and triggered an event, but all the "reports" I created manually are put straight into draft state.

**Reply by captn3m0** [reply]
An advisory is different from a "report", the latter is an externally submitted advisory of sorts. If you use your own account, you don't see the report form at all, and GitHub forces you to directly create an advisory.

I've filed a report at https://github.com/sethmlarson/ghsa-default-collabs/security/advisories/GHSA-w5vc-rwc4-w2f3. Did the event fire? If not, I could give it a shot.

**Reply by sethmlarson** [reply]
@captn3m0 Thanks for doing that manually, indeed you are correct the event only fires for the "report" flow and not the "draft" flow.

**Reply by sethmlarson** [reply]
@captn3m0 I've created a feature request for such a feature: https://github.com/orgs/community/discussions/67871
